server/src/main/java/org/elasticsearch/action/admin/cluster/node/reload/TransportNodesReloadSecureSettingsAction.java

```diff
@@ -145,7 +145,13 @@ protected NodesReloadSecureSettingsResponse.NodeResponse nodeOperation(NodeReque
         } catch (final Exception e) {
             return new NodesReloadSecureSettingsResponse.NodeResponse(clusterService.localNode(), e);
         } finally {
-            secureSettingsPassword.close();
+            // Do not close the secureString if this is executed on the local node. It is shared with
+            // the Nodes level request, it will be closed when the nodes request is completed. Since it
+            // is also shared by all Node level requests, closing it here can sometimes cause serialization
+            // failure when sending other non-local-node requests across the wire.
+            if (false == clusterService.localNode().getId().equals(nodeReloadRequest.getParentTask().getNodeId())) {
+                secureSettingsPassword.close();
+            }
```

Comment:
This feels like quite a hack. It implies we don't really know where the resource ownership is supposed to be. 

Would it be possible to solve this by cloning the secure string so that we can always close it?

Reply:
That was my initial thought as well. But it requires:
1. Either handling BWC for serialisation or replicate NodesRequest serialisation logic inside NodeRequest and just replace secureString one with the clone.
2. Having a way to actively close the cloned secret for each  individual Node level request after it is sent across the wire. 

Both are doable, but it is a larger change than what I had hoped for. I can make these changes if you prefer.

Reply:
Just pushed new changes to clone the secret at the sender side. The Nodes level request now also track the node level requests so they can be all closed together after request is complete. Please let me know if this works for you.